node/src/components/consensus/highway_core/state/panorama.rs

```diff
@@ -236,3 +237,63 @@ impl<C: Context> Panorama<C> {
         Ok(())
     }
 }
+
+/// Utility enum to represent participation of the validator in the panorama.
+pub(crate) enum ParticipationStatus {
+    Active,
+    None,
+    Faulty,
+}
+
+impl Debug for ParticipationStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ParticipationStatus::Active => write!(f, "A"),
+            ParticipationStatus::None => write!(f, "N"),
+            ParticipationStatus::Faulty => write!(f, "F"),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct PanoramaStakes {
+    validators_participation: ValidatorMap<ParticipationStatus>,
+    era_weight: Weight,
+    active_weight: Weight,
+    passive_weight: Weight,
+    faulty_weight: Weight,
+}
+
+impl PanoramaStakes {
+    pub(crate) fn new<C: Context>(panorama: &Panorama<C>, weights: &ValidatorMap<Weight>) -> Self {
+        let mut active_weight = Weight(0);
+        let mut passive_weight = Weight(0);
+        let mut faulty_weight = Weight(0);
+        let panorama_voted_stake: Vec<ParticipationStatus> = panorama
+            .enumerate()
+            .map(|(idx, obs)| match obs {
+                Observation::Correct(_) => {
+                    active_weight += *weights.get(idx);
+                    ParticipationStatus::Active
+                }
+                Observation::Faulty => {
+                    faulty_weight += *weights.get(idx);
+                    ParticipationStatus::Faulty
+                }
+                Observation::None => {
+                    passive_weight += *weights.get(idx);
+                    ParticipationStatus::None
+                }
+            })
+            .collect();
+        let validators_participation = ValidatorMap::from(panorama_voted_stake);
```

Comment:
I think this can be [`collect`ed directly as a `ValidatorMap`](https://github.com/CasperLabs/casper-node/blob/9056ac7e8d93d84a5e3e61825a29eb2f0e22440a/node/src/components/consensus/highway_core/validators.rs#L185), instead of allocating a `Vec` first.